### ISSUE TITLE
Use TriggerDagRunOperator instead of custom trigger dag function?

### ISSUE THREAD

**Issue by greenape** [Flowminder/FlowKit#984]
I'm just wondering if, post #978 merging we couldn't use airflow's built in  [TriggerDagRunOperator](https://airflow.apache.org/_api/airflow/operators/dagrun_operator/index.html) instead of the custom function to trigger the dag from the sensor?

**Comment by danwilliams** [comment]
This does not seem to be feasible/possible at the moment.

After a number of experiments locally, this doesn't seem to work nicely as we'd expect. Because, in the `production_trigger__callable` we're determining which sources to process then triggering the specific dag run for each unprocessed source, there doesn't seem to be a way to use the `TriggerDagRunOperator` without refactoring the existing `etl_sensor` DAG to always process the sources and for the "latest" (or similar) unprocessed source, trigger the specific DAG, and end up triggering the sensor multiple times to get all the unprocessed sources (which would be a performance loss).

Basically, `TriggerDagRunOperator` is good for running a single instance of a single DAG, while we need multiple instances, of multiple DAGs - we could get away with multiple DAGs (since we know _which_ are the DAGs) but single instance by refactoring as above but not sure that's worth it from a performance point of view currently.
